API/front.py

```diff
@@ -0,0 +1,27 @@
+from flask import render_template, Blueprint
+import requests
+import json
+from datetime import datetime
+
+front = Blueprint('front', __name__, template_folder='templates')
+@front.route('/')
+def template_test():
+    # We get the user information from our mocked API
+    user = requests.get('https://hackathoooon.osc-fr1.scalingo.io/mock/pass/calendar?user=1')
+    user_calendar = json.loads(user.text)['calendar']
+
+    # We search the next_course
+    today = datetime.today()
+    today_course = ""
+    while today_course == "":
+        try:
+            # We try to find a course for today, if we don't find => fail
+            today_course = user_calendar[today.strftime('%A')]
+            break
+        except:
+            # We pass to the next day
+            today += datetime.timedelta(days=1)
+            break
```

Comment:
Pourquoi un `try/except` ici ? S'il a une réelle utilité, il vaut mieux voir précisément quelle exception tu essaie de catch et l'indiquer dans le `except`, autrement avec ça tu catch toutes les exceptions sans les traiter correctement derrière, ça peut engendrer des problèmes potentiels.

Par ailleurs, pourquoi tu break dans ton except ? Actuellement ton `while` sert à rien puisque dans tous les cas tu break à la première itération.

Reply:
Enfaite au début je voulais rechercher le cours suivant dans le calendrier. Dans la journée ou le prochain dans les jours suivants.

Mais je ne voyais pas trop comment parcourir les horaire en python autrement qu'avec while, puis pour le prototype, je me suis dis qu'on prenais juste le premier cours de la journée pour la prochaine journée qui a un cours.

Reply:
Utilise ça :
```python
def get_current_or_next_course():
    user_calendar = json.loads(...)['calendar']

    # We search for the next course
    now = datetime.now() + timedelta(days=1)
    today = now.strftime('%A')

    # First try to get the courses of today
    today_courses = user_calendar.get(today, [])

    if today_courses:
        # We have one or more classes this day
        next_or_current_courses = list(filter(
            lambda course: now < datetime.strptime(course['end'], '%H:%M'),
            today_courses
        ))
        if next_or_current_courses:
            # Return the next or current course left in the day
            return next_or_current_courses[0]

    next_day = now
    while next_day < next_day + timedelta(days=7):
        next_day += timedelta(days=1)
        this_day_courses = user_calendar.get(next_day.strftime('%A'), [])
        if this_day_courses:
            return this_day_courses[0]

    return {}
```